labview/msg_headers.c

```diff
@@ -0,0 +1,270 @@
+          headerBuffer +
+          headerBufferLen;  // If no delimiter found, use the end of the string
+    }
+    if (delimiter == headerBuffer + headerBufferLen) {
+      finished = TRUE;
+    }
+    // Find the equal_char '='
+    char* equal_char = findInBuffer(headerBuffer, headerBufferLen, index, 0x3D);
+
+    if (equal_char != NULL && equal_char < delimiter) {
+      // Add the entry to the entries array
+      amqp_table_entry_t* newEntries =
+          realloc(entries, (numEntries + 1) * sizeof(amqp_table_entry_t));
+      if (newEntries == NULL) {
+        // Error handling for memory allocation failure
+        perror("Memory allocation failed");
+        free(entries);
+        return table;
+      }
+      entries = newEntries;
+      amqp_field_value_t value;
+
+      // Get the kind (first byte)
+      value.kind = headerBuffer[index];
+      index++;  // Move to the next character
+
+      // Parse the key
+      entries[numEntries].key.bytes = headerBuffer + index;
+      entries[numEntries].key.len = equal_char - ((char*)headerBuffer + index);
+
+      // Init and parse the value
+      char valueTempBuffer[8];
+      int valueLength = delimiter - equal_char - 1;
+      memcpy(valueTempBuffer, equal_char + 1, valueLength);
+
+      // Parse the value based on the kind
+      switch (value.kind) {
+        case AMQP_FIELD_KIND_UTF8:
+        case AMQP_FIELD_KIND_BYTES:
+          value.value.bytes.bytes = equal_char + 1;  // fist char after "="
+          value.value.bytes.len = valueLength;
+          break;
+        case AMQP_FIELD_KIND_I8:
+          value.value.i8 = (int8_t)valueTempBuffer[0];
+          break;
+        case AMQP_FIELD_KIND_U8:
+          value.value.u8 = (uint8_t)valueTempBuffer[0];
+          break;
+        case AMQP_FIELD_KIND_I16:
+          value.value.i16 = *((int16_t*)valueTempBuffer);
+          break;
+        case AMQP_FIELD_KIND_U16:
+          value.value.u16 = *((uint16_t*)valueTempBuffer);
+          break;
+        case AMQP_FIELD_KIND_I32:
+          value.value.i32 = *((int32_t*)valueTempBuffer);
+          break;
+        case AMQP_FIELD_KIND_U32:
+          value.value.u32 = *((uint32_t*)valueTempBuffer);
+          break;
+        case AMQP_FIELD_KIND_I64:
+          value.value.i64 = *((int64_t*)valueTempBuffer);
+          break;
+        case AMQP_FIELD_KIND_U64:
+        case AMQP_FIELD_KIND_TIMESTAMP:
+          value.value.u64 = *((uint64_t*)valueTempBuffer);
+          break;
+        case AMQP_FIELD_KIND_F32:
+          value.value.f32 = *((float*)valueTempBuffer);
+          break;
+        case AMQP_FIELD_KIND_F64:
+          value.value.f64 = *((double*)valueTempBuffer);
+          break;
+        default:
+          // Unsupported kind, ignore this header
+          break;
+      }
+
+      entries[numEntries].value = value;
+      numEntries++;
+    }
+
+    // Move to the next header (skip the delimiter ';')
+    index = delimiter - (char*)headerBuffer + 1;
```

Comment:
can you explain why it is not index=delimiter+1?

Reply:
sure, the index is a variable that indicates an element in the array so index = 3 will indicate the array[3]. While delimiter and (char*)header buffer are pointers. I prefer to have a delimiter as a pointer because it makes it possible to use it in memory copy functions.